design-proposals/edge-infra-day2-improvements.md

```diff
@@ -0,0 +1,357 @@
+available updates for the Edge Node.
+
+The following represents the schema changes to be applied to EIM schema to 
+fulfill the above requirements:
+
+![Edge Infrastructure Manager Day2 Entity Relationship Diagram](./schema_day2_changes.drawio.svg)
+
+Changes:
+- **OS Profile**:
+  - `installed_packages`: field is used to track only packages installed at 
+    day 0, for Immutable those comes from the Manifest file published 
+    together with the EMT image. For Mutable, those are filled using the 
+    Ubuntu manifest file. For Mutable, this won't include Bare Metal Agents 
+    packages, since those are installed during Day 0, but they are not part 
+    of the Ubuntu manifest.
+  - `update_sources`: the field is deprecated, and should not be used anymore.
+  - The resource now supports only Create, Read and Delete operations. Update 
+    is not allowed anymore. The resource is now read-only from the 
+    Northbound APIs perspective.
+- **Instance**:
+  - `desired_os`: field is removed, we don't drive the Immutable OS day 2 
+    workflow from this field anymore.
+  - `current_os`: the field can now be changed from EIM northbound APIs 
+    (REST), only before the EN is provisioned. After that, the field is 
+    handled internally by EIM, and will be updated based on the current OS 
+    installed.
+  - `installed_packages`: this field is used to track packages that are 
+    actually installed in the Edge Node at runtime. For Immutable OSes, we 
+    expect this to be equal to the content of `installed_packages` in the 
+    OS Profile. For Mutable OSes, this could change during the lifetime of 
+    the Edge Node.
+  - `new_versions`: signals that new updates are available. This field doesn't 
+    tell anything about the fact that the update will be done in the next 
+    scheduled maintenance. The field is a string.
+    - mutable: list of packages ready to be upgraded on the Edge Node, 
+      empty if no new versions are available.
+    - immutable: latest EMT version, empty if the version is already latest.
+  - `next_update_policy`: The update policy that is applied during the 
+    maintenance schedule.
+  
+New Resources:
+
+- **OS Update Policy**: defines the Policy that must be applied during an 
+  update. This resource can be linked to multiple Instances, Regions and or 
+  Sites. Some fields and edges are valid for Mutable or Immutable OSes.
+  - `install_packages`: list of packages to be installed during the update. 
+    This field is used only for Mutable OSes.
+  - `update_sources`: 'DEB822 Source Format' for Debian style OSs, contains 
+    the sources of the packages to be installed during the update, if new 
+    sources needs to be added.
+  - `update_policy`: enum field, defines which policy to use:
+    - `UPDATE_POLICY_LATEST`: upgrade to latest version. 
+      - mutable: upgrade all packages to latest versions.
+      - immutable: upgrade the OS version to the latest.
```

Comment:
Does `latest` mean the version in `target_os` for immutable OSs?

Because OS Update Policy is update-only does a new one need to be created/associated to instances every time the `target_os` changes?

Reply:
The idea is to have POLICY_LATEST: use the latest version following semver. POLICY_TARGET: use the targetOS, or install_package.
Let me clarify it.

Reply:
Let me know if now is clear enough